### ISSUE TITLE
improve user journey for scripting

### ISSUE THREAD

**Issue by a-wallen** [sherlock-project/sherlock#2121]
## Checklist

- [X] I'm reporting a feature request
- [X] I've checked for similar feature requests including closed ones

## Description

Sherlock is great from the CLI, but not as much trying to write a script using it, here's why.

1. The documentation does not have clear instructions for calling the `sherlock` function with it's required arguments. Here's an [example](https://github.com/sherlock-project/sherlock/blob/f2090bd19806cece29b0acc9fe7801ef6230856e/sherlock/sherlock.py#L175-L177). The instructions do not clearly state how to provide a default argument, stating how to find the class that needs to be implemented.  
2. Just like the CLI tool, the `sherlock` function should be able to work with just the `username` argument. For example: the `site_data` could be defaulted to data.json. 
3. I don't want to suggest that the `query_notifier` parameter gets defaulted. It's fine the way that it is. However, there should be a separate implementation called `QueryNotifierMap` where you can access `result` as a map. `QueryNotifyPrint` is for the CLI, developers want to call this with just an object.

**Comment by matheusfelipeog** [comment]
Hi @a-wallen, thanks for opening this issue ;)

Sherlock is essentially a tool focused on being used via CLI, which is why there isn't a more comprehensive documentation of its internal components.

I personally like the idea of providing clear and documented ways to integrate Sherlock into other projects; that's actually a good idea. We can improve on that over time, although it's not one of our priorities right now.

**Comment by ppfeister** [comment]
Sherlock is packaged in several places and isn't likely to be ran as a single _script_ --- it's more likely that people will install the package itself via pip or similar

When using the package, people aren't calling the sherlock function. They're calling the sherlock _package_. In order to call the sherlock _function_ it'd have to be something like

```python
from sherlock import sherlock # module within package
sherlock.sherlock(usernames) # function within module
```

quite verbose when

```python
import sherlock # whole package
sherlock(usernames) # using package entry point
```

is probably preferred in 99% of cases.

I believe main() is the current package entry point. main() should currently take a ton of parsed sys args rather than function args. Would you be able to somehow adapt it so the entry point accepts _both_ command line args _and_ normal scripted function args? How difficult do you think that adaptation would be?

**Comment by a-wallen** [comment]
@ppfeister yup you're on point: this is what I think you mean https://github.com/sherlock-project/sherlock/pull/2143
